### ISSUE TITLE
Not liked: 403 ERROR

### ISSUE THREAD

**Issue by eosman12** [instabot-py/instabot.py#1587]
Check this or we will delete your issue. (fill in the checkbox with an X like so: [x])
- [X] I have searched for other issues with the same problem. (including #1153)

## Hosting
- [ ] Are you hosting the bot on a VPS or other Cloud hosting?
- [X] If so; have you tried running the code on your local machine?

#### Select one:
- [X] Bug
- [ ] Feature Request
- [X] Technical Help
- [ ] Other

### Operating System? (include version)
- [X] MAC
- [ ] Windows
- [ ] Linux
- [ ] Android

## Python Version?
- [ ] 2.x
- [X] 3.x

## Description of your issue
Log in is successful but getting 403 error and bot isn't doing anything
I have spent hours reading through github forums and applying fixes but nothing has worked, please help!
It loops in a 403 error, I've tried [#1498](https://github.com/instabot-py/instabot.py/pull/1498) with no luck.
```
10.09.2018_23:51 Trying to like media: 1865604621555018600
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865593610666280620
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865591927921383251
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865588460112810150
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865587399616443042
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865579778045093732
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865569994186502468
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865568743159430259
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865559667339387074
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865554429484713197
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865553452948670105
10.09.2018_23:51 Not liked: 403
10.09.2018_23:51 Trying to like media: 1865551889698318856
10.09.2018_23:51 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865549592781546885
10.09.2018_23:52 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865549634463514068
10.09.2018_23:52 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865546458251868947
10.09.2018_23:52 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865545549690147674
10.09.2018_23:52 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865542795442235255
10.09.2018_23:52 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865542338799867411
10.09.2018_23:52 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865533453603157432
10.09.2018_23:52 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865468193235868217
10.09.2018_23:52 Not liked: 403
10.09.2018_23:52 Trying to like media: 1865526378825600103
10.09.2018_23:52 Not liked: 403
```

## Config

```
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import os
import time

from src import InstaBot
from src.check_status import check_status
from src.feed_scanner import feed_scanner
from src.follow_protocol import follow_protocol
from src.unfollow_protocol import unfollow_protocol

bot = InstaBot(
    login="XXX",
    password="XXX",
    like_per_day=1000,
    comments_per_day=0,
    tag_list=['my_tags'],
    tag_blacklist=['rain', 'thunderstorm'],
    user_blacklist={},
    max_like_for_one_tag=50,
    follow_per_day=300,
    follow_time=2 * 60,
    unfollow_per_day=0,
    unfollow_break_min=15,
    unfollow_break_max=30,
    log_mod=0,
    proxy='',
    # List of list of words, each of which will be used to generate comment
    # For example: "This shot feels wow!"
    comment_list=[["this", "the", "your"],
                  ["photo", "picture", "pic", "shot", "snapshot"],
                  ["is", "looks", "feels", "is really"],
                  ["great", "super", "good", "very good", "good", "wow",
                   "WOW", "cool", "GREAT","magnificent", "magical",
                   "very cool", "stylish", "beautiful", "so beautiful",
                   "so stylish", "so professional", "lovely",
                   "so lovely", "very lovely", "glorious","so glorious",
                   "very glorious", "adorable", "excellent", "amazing"],
                  [".", "..", "...", "!", "!!", "!!!"]],
    # Use unwanted_username_list to block usernames containing a string
    ## Will do partial matches; i.e. 'mozart' will block 'legend_mozart'
    ### 'free_followers' will be blocked because it contains 'free'
    unwanted_username_list=[
        'second', 'stuff', 'art', 'project', 'love', 'life', 'food', 'blog',
        'free', 'keren', 'photo', 'graphy', 'indo', 'travel', 'art', 'shop',
        'store', 'sex', 'toko', 'jual', 'online', 'murah', 'jam', 'kaos',
        'case', 'baju', 'fashion', 'corp', 'tas', 'butik', 'grosir', 'karpet',
        'sosis', 'salon', 'skin', 'care', 'cloth', 'tech', 'rental', 'kamera',
        'beauty', 'express', 'kredit', 'collection', 'impor', 'preloved',
        'follow', 'follower', 'gain', '.id', '_id', 'bags'
    ],
    unfollow_whitelist=['example_user_1', 'example_user_2'])
while True:

    #print("# MODE 0 = ORIGINAL MODE BY LEVPASHA")
    #print("## MODE 1 = MODIFIED MODE BY KEMONG")
    #print("### MODE 2 = ORIGINAL MODE + UNFOLLOW WHO DON'T FOLLOW BACK")
    #print("#### MODE 3 = MODIFIED MODE : UNFOLLOW USERS WHO DON'T FOLLOW YOU BASED ON RECENT FEED")
    #print("##### MODE 4 = MODIFIED MODE : FOLLOW USERS BASED ON RECENT FEED ONLY")
    #print("###### MODE 5 = MODIFIED MODE : JUST UNFOLLOW EVERYBODY, EITHER YOUR FOLLOWER OR NOT")

    ################################
    ##  WARNING   ###
    ################################

    # DON'T USE MODE 5 FOR A LONG PERIOD. YOU RISK YOUR ACCOUNT FROM GETTING BANNED
    ## USE MODE 5 IN BURST MODE, USE IT TO UNFOLLOW PEOPLE AS MANY AS YOU WANT IN SHORT TIME PERIOD

    mode = 0

    #print("You choose mode : %i" %(mode))
    #print("CTRL + C to cancel this operation or wait 30 seconds to start")
    #time.sleep(30)

    if mode == 0:
        bot.new_auto_mod()

    elif mode == 1:
        check_status(bot)
        while bot.self_following - bot.self_follower > 200:
            unfollow_protocol(bot)
            time.sleep(10 * 60)
            check_status(bot)
        while bot.self_following - bot.self_follower < 400:
            while len(bot.user_info_list) < 50:
                feed_scanner(bot)
                time.sleep(5 * 60)
                follow_protocol(bot)
                time.sleep(10 * 60)
                check_status(bot)

    elif mode == 2:
        bot.bot_mode = 1
        bot.new_auto_mod()

    elif mode == 3:
        unfollow_protocol(bot)
        time.sleep(10 * 60)

    elif mode == 4:
        feed_scanner(bot)
        time.sleep(60)
        follow_protocol(bot)
        time.sleep(10 * 60)

    elif mode == 5:
        bot.bot_mode = 2
        unfollow_protocol(bot)

    else:
        print("Wrong mode!")
```

**Comment by TractorNyx** [comment]
I used the solution topic (https://github.com/instabot-py/instabot.py/pull/1498) for error 403, but still it still shows the error "Not liked 403" :(

**Comment by GioZaarour** [comment]
I've been getting the same problem for weeks and I haven't found a workaround. Is it maybe instagram blocking the like attempts?????? Nothing works even with a VPN. Please help!!!!!

**Comment by eosman12** [comment]
I used the instabot.py file on this link and it’s working
https://github.com/erikpi/instabot.py/find/master?q=

**Comment by sjamod9** [comment]
I can confirm this issue only occurs in Mac OS.
After applying https://github.com/instabot-py/instabot.py/commit/806e09414f73a461bdb81fd6d4fc2e1e34a54ef8 and https://github.com/instabot-py/instabot.py/pull/1503/files Can somebody help us to fix this issue!

**Comment by ZVesta** [comment]
Do you guys know if I can use the program simultaneously on 2 accounts at the same time? My first one runs perfectly with #1498 fix but when I run the second file I created (example1.py) it logs in but gives me 403 error :-/ 
Windows 10.

**Comment by pinkoveralls** [comment]
@ZVesta With your main file you're not getting 403 errors? Did you apply any other fixes besides that one to the main files? I'm just getting continuous 403 errors as of a couple days ago. 

Not sure if you can run it on 2 accounts at the same time on a tingle terminal, you might try running it outside your terminal for the second account.

**Comment by ZVesta** [comment]
@pinkoveralls you know, I don't remember 100% if I applied other fixes cause it was a couple of weeks ago. It took me a while to make it fly but then it worked with no errors. It is still running as I write this message.
However I copied example.py to an example1.py with new login info and some settings like tags and tried running it in a new terminal window. It logs in but gives me 403. I played around it (renamed it example.py , left everything totally unchanged except for login and password) but nothing helps. I didn't try to close my first terminal window since I am afraid it will break too and I don't want it. I guess here I must mention I am not good at coding, I just took one course on edx.com which let me understand some of it and write basic stuff :) 

By the way, I could only run my program in mode 0, all others give me errors. Did that work for you?

**Comment by pinkoveralls** [comment]
@zvesta Mine stopped working about two days ago, and some people seems to be having the same problems but for others it works. I think it's because instabot hasn't been updated in two months and people applied different PR/fixes to make theirs work, and now some seem to be broken while others seem to be working. I'm hoping somebody with a working bot will remember which fixes they used, or it will be updated properly. 

I only run it in mode 0 since I don't use it to follow very often, so I've never tried it in any other mode.

**Comment by ZVesta** [comment]
@pinkoveralls Let's hope someone with better knowledge and understanding will be able to fix that :D 
Dear god, I should have studied IT, not social sciences!

**Comment by aevrisso** [comment]
@ZVesta Si puedes ejecutar varias instancias, solo que debes cambiar la base de datos en cada ejecución, es decir, si tienes una cuenta "cuentadeejemplo" la base de datos debes nombrarla como "cuentadeejemplo", si tienes otra cuenta "cuenta2" debes nombrarla como "cuenta2". 

### @ZVesta If you can execute multiple instances, you only have to change the database in each execution, that is, if you have an account "cuentadeejemplo" the database must name it as "cuentaemjemplo", if you have another account "cuenta2" you must name it as "account2".
Yo tengo una carpeta por cuenta, solo que cambie en mi instabot.py 

> database_name = "nombredelacuenta.db"
Notaras que después de este cambio, python agregará una nueva db o un nuevo archivo .db en tu carpeta principal.

### You will notice that after this change, python will add a new db or a new .db file in your main folder.

Sin embargo, también puedes cambiar:
### However, you can also change:

> database_name =""

dentro de tu archivo example.py
### inside your example.py file
>     login="login",
>     password="password",
>     database_name = "follows_nombredelacuenta.db",

**Comment by ZuxGinger** [comment]
> r = self.s.get(self.url) csrf_token = re.search('(?<=\"csrf_token\":\")\w+', r.text).group(0) self.s.headers.update({'X-CSRFToken': r.cookies['csrftoken']}) time.sleep(5 * random.random()) login = self.s.post(self.url_login, data=self.login_post, allow_redirects=True) self.s.headers.update({'X-CSRFToken': login.cookies['csrftoken']}) self.csrftoken = login.cookies['csrftoken'] #ig_vw=1536; ig_pr=1.25; ig_vh=772; ig_or=landscape-primary;

Solved my latest problem on MacOS.

**Comment by ZuxGinger** [comment]
> Hi i change this:
> 
> > ```
> >     r = self.s.get(self.url)
> >     csrf_token = re.search('(?<=\"csrf_token\":\")\w+', r.text).group(0)
> >     self.s.headers.update({'X-CSRFToken': csrf_token})
> >     time.sleep(5 * random.random())
> >     login = self.s.post(self.url_login, data=self.login_post, allow_redirects=True)
> >     self.csrftoken = csrf_token
> >     # ig_vw=1536; ig_pr=1.25; ig_vh=772;  ig_or=landscape-primary;
> > ```
> 
> _****************************************FIXED****************************************_____
> 
> > ```
> >    r = self.s.get(self.url)
> >     csrf_token = re.search('(?<=\"csrf_token\":\")\w+', r.text).group(0)
> >     self.s.headers.update({'X-CSRFToken': r.cookies['csrftoken']})
> >     time.sleep(5 * random.random())
> >     login = self.s.post(self.url_login, data=self.login_post, allow_redirects=True)
> >     self.s.headers.update({'X-CSRFToken': login.cookies['csrftoken']})
> >     self.csrftoken = login.cookies['csrftoken']
> >     #ig_vw=1536; ig_pr=1.25; ig_vh=772;  ig_or=landscape-primary;
> > ```
> 
> He probado esto y me funciona perfectamente ahora, yo tenia el error 403 (not liked), tampoco me comentaba ni seguía, pero ya funciona. gracias a @erikpi
> 
> I have tried this and it works perfectly now, I had the error 403 (I do not like it), I did not comment or follow, but it works. thanks to @erikpi

Solved my latest problem on MacOS.

**Comment by jfinly** [comment]
> Hi i change this:
> 
> > ```
> >     r = self.s.get(self.url)
> >     csrf_token = re.search('(?<=\"csrf_token\":\")\w+', r.text).group(0)
> >     self.s.headers.update({'X-CSRFToken': csrf_token})
> >     time.sleep(5 * random.random())
> >     login = self.s.post(self.url_login, data=self.login_post, allow_redirects=True)
> >     self.csrftoken = csrf_token
> >     # ig_vw=1536; ig_pr=1.25; ig_vh=772;  ig_or=landscape-primary;
> > ```
> 
> _****************************************FIXED****************************************_____
> 
> > ```
> >    r = self.s.get(self.url)
> >     csrf_token = re.search('(?<=\"csrf_token\":\")\w+', r.text).group(0)
> >     self.s.headers.update({'X-CSRFToken': r.cookies['csrftoken']})
> >     time.sleep(5 * random.random())
> >     login = self.s.post(self.url_login, data=self.login_post, allow_redirects=True)
> >     self.s.headers.update({'X-CSRFToken': login.cookies['csrftoken']})
> >     self.csrftoken = login.cookies['csrftoken']
> >     #ig_vw=1536; ig_pr=1.25; ig_vh=772;  ig_or=landscape-primary;
> > ```
> 
> He probado esto y me funciona perfectamente ahora, yo tenia el error 403 (not liked), tampoco me comentaba ni seguía, pero ya funciona. gracias a @erikpi
> 
> I have tried this and it works perfectly now, I had the error 403 (I do not like it), I did not comment or follow, but it works. thanks to @erikpi

where is this change made?

**Comment by aevrisso** [comment]
> where is this change made?
instabot.py
@jfinly

**Comment by jfinly** [comment]
> Solved [e3a935d](https://github.com/instabot-py/instabot.py/commit/e3a935d13a4a8299d39353b7d23751d233cde3ee)

That doesn't solve anything; I've had that change for a week or longer.

**Comment by alphabetagamer** [comment]
> Hi i change this:
> 
> > ```
> >     r = self.s.get(self.url)
> >     csrf_token = re.search('(?<=\"csrf_token\":\")\w+', r.text).group(0)
> >     self.s.headers.update({'X-CSRFToken': csrf_token})
> >     time.sleep(5 * random.random())
> >     login = self.s.post(self.url_login, data=self.login_post, allow_redirects=True)
> >     self.csrftoken = csrf_token
> >     # ig_vw=1536; ig_pr=1.25; ig_vh=772;  ig_or=landscape-primary;
> > ```
> 
> _****************************************FIXED****************************************_____
> 
> > ```
> >    r = self.s.get(self.url)
> >     csrf_token = re.search('(?<=\"csrf_token\":\")\w+', r.text).group(0)
> >     self.s.headers.update({'X-CSRFToken': r.cookies['csrftoken']})
> >     time.sleep(5 * random.random())
> >     login = self.s.post(self.url_login, data=self.login_post, allow_redirects=True)
> >     self.s.headers.update({'X-CSRFToken': login.cookies['csrftoken']})
> >     self.csrftoken = login.cookies['csrftoken']
> >     #ig_vw=1536; ig_pr=1.25; ig_vh=772;  ig_or=landscape-primary;
> > ```
> 
> He probado esto y me funciona perfectamente ahora, yo tenia el error 403 (not liked), tampoco me comentaba ni seguía, pero ya funciona. gracias a @erikpi
> 
> I have tried this and it works perfectly now, I had the error 403 (I do not like it), I did not comment or follow, but it works. thanks to @erikpi

this works
